src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapException.cs

```diff
@@ -36,32 +35,33 @@ internal enum LdapError
 
     internal class LdapErrorMappings
     {
-        private static Hashtable s_resultCodeHash = null;
+        private static Dictionary<LdapError, string> s_resultCodeHash = null;
 
         static LdapErrorMappings()
         {
-            s_resultCodeHash = new Hashtable();
-
-            s_resultCodeHash.Add(LdapError.IsLeaf, String.Format(CultureInfo.CurrentCulture, SR.LDAP_IS_LEAF));
-            s_resultCodeHash.Add(LdapError.InvalidCredentials, String.Format(CultureInfo.CurrentCulture, SR.LDAP_INVALID_CREDENTIALS));
-            s_resultCodeHash.Add(LdapError.ServerDown, String.Format(CultureInfo.CurrentCulture, SR.LDAP_SERVER_DOWN));
-            s_resultCodeHash.Add(LdapError.LocalError, String.Format(CultureInfo.CurrentCulture, SR.LDAP_LOCAL_ERROR));
-            s_resultCodeHash.Add(LdapError.EncodingError, String.Format(CultureInfo.CurrentCulture, SR.LDAP_ENCODING_ERROR));
-            s_resultCodeHash.Add(LdapError.DecodingError, String.Format(CultureInfo.CurrentCulture, SR.LDAP_DECODING_ERROR));
-            s_resultCodeHash.Add(LdapError.TimeOut, String.Format(CultureInfo.CurrentCulture, SR.LDAP_TIMEOUT));
-            s_resultCodeHash.Add(LdapError.AuthUnknown, String.Format(CultureInfo.CurrentCulture, SR.LDAP_AUTH_UNKNOWN));
-            s_resultCodeHash.Add(LdapError.FilterError, String.Format(CultureInfo.CurrentCulture, SR.LDAP_FILTER_ERROR));
-            s_resultCodeHash.Add(LdapError.UserCancelled, String.Format(CultureInfo.CurrentCulture, SR.LDAP_USER_CANCELLED));
-            s_resultCodeHash.Add(LdapError.ParameterError, String.Format(CultureInfo.CurrentCulture, SR.LDAP_PARAM_ERROR));
-            s_resultCodeHash.Add(LdapError.NoMemory, String.Format(CultureInfo.CurrentCulture, SR.LDAP_NO_MEMORY));
-            s_resultCodeHash.Add(LdapError.ConnectError, String.Format(CultureInfo.CurrentCulture, SR.LDAP_CONNECT_ERROR));
-            s_resultCodeHash.Add(LdapError.NotSupported, String.Format(CultureInfo.CurrentCulture, SR.LDAP_NOT_SUPPORTED));
-            s_resultCodeHash.Add(LdapError.NoResultsReturned, String.Format(CultureInfo.CurrentCulture, SR.LDAP_NO_RESULTS_RETURNED));
-            s_resultCodeHash.Add(LdapError.ControlNotFound, String.Format(CultureInfo.CurrentCulture, SR.LDAP_CONTROL_NOT_FOUND));
-            s_resultCodeHash.Add(LdapError.MoreResults, String.Format(CultureInfo.CurrentCulture, SR.LDAP_MORE_RESULTS_TO_RETURN));
-            s_resultCodeHash.Add(LdapError.ClientLoop, String.Format(CultureInfo.CurrentCulture, SR.LDAP_CLIENT_LOOP));
-            s_resultCodeHash.Add(LdapError.ReferralLimitExceeded, String.Format(CultureInfo.CurrentCulture, SR.LDAP_REFERRAL_LIMIT_EXCEEDED));
-            s_resultCodeHash.Add(LdapError.SendTimeOut, String.Format(CultureInfo.CurrentCulture, SR.LDAP_SEND_TIMEOUT));
+            s_resultCodeHash = new Dictionary<LdapError, string>
```

Comment:
Same comments as previously, including that it looks like the type change could result in previously null-returning behavior now resulting in exceptions.